resources/persist/src/lib.rs

```diff
@@ -41,6 +47,41 @@ impl PersistInstance {
         Ok(serialize_into(&mut writer, &struc).map_err(PersistError::Serialize))?
     }
 
+    /// list method returns a vector of strings containing all the keys associated with a PersistInstance
+    pub fn list(&self) -> Result<Vec<String>, PersistError> {
+        let storage_folder = self.get_storage_folder();
+
+        let mut list = Vec::new();
+
+        let entries = fs::read_dir(storage_folder).map_err(PersistError::ListFolder)?;
+        for entry in entries {
+            let dir = entry.map_err(PersistError::ListFolder)?;
+            list.push(
+                dir.path()
+                    .into_os_string()
+                    .into_string()
+                    .unwrap_or_else(|_| "path contains non-UTF-8 characters".to_string())
+                    .split(".bin")
```

Comment:
Splitting here works, but can break: if a key `aaa.binbbb` is saved, that would make the file `aaa.binbbb.bin`, which after this split+collect becomes `aaabbb`. I would say trimming the file extension from the end is better. You can check out `std::path::Path::with_extension`.

Reply:
Struggling here.  Not seeing how std:::path::Path::with_extension helps. I'm getting better at interpreting the docs content, but I have a ways to go. Will resume the head banging.

Reply:
My mistake. The `dir.path()` method gives you a PathBuf, and there the method is called `set_extension()`. The idea is to modify the path of the dir entry before converting it to a string, making it more "properly" handled.

Reply:
Have adjusted the list method to a good form now, I feel.